parser.go

```diff
@@ -1077,8 +1077,9 @@ func (p *parser) parseSchemaObject(pkgPath, pkgName, typeName string) (*SchemaOb
 			// p.debugf("guess %s ast.TypeSpec in package %s", guessTypeName, guessPkgName)
 			typeSpec, exist = p.getTypeSpec(guessPkgPath, guessPkgName, guessTypeName)
 			if !exist {
-				p.debugf("can not find definition of guess %s ast.TypeSpec in package %s", guessTypeName, guessPkgName)
-				return &schemaObject, nil
+				log.Fatalf("Cannot find definition of guess %s ast.TypeSpec in package %s. " +
```

Comment:
instead of logging and fatal-ing,  could return an err instead? Then we'd be able to easily unit test it by asserting that an error was expected.

Reply:
Possibly. This would be a much more involved change, this function is called all over the place with this pattern of swallowing the error, for example:
```
schema, err := p.parseSchemaObject(pkgPath, pkgName, goType)
if err != nil {
    p.debug("parseResponseComment: cannot parse goType", goType)
}
```
So I went with the pattern already in use. Would this do as a first pass?

Reply:
oof. yeah if it's too hard to surface the error, this is fine.

Reply:
for any improvements to goas we don't have time for now, can we make CH tickets for them? we might want to create a `goas` project in Clubhouse actually; it's something the dev exp squad should own in the future